docs/docs/dev_docs/tutorials/token_portal/minting_on_aztec.md

```diff
@@ -0,0 +1,96 @@
+---
+title: Minting tokens on Aztec
+---
+
+In this step we will start writing our Aztec.nr bridge smart contract!
+
+## Consume the L1 message
+
+We have now moved our funds to the portal and created a L1->L2 message. Upon building the next rollup, the sequencer asks the inbox for any incoming messages and adds them to Aztec’s L1->L2 message tree, so an application on L2 can prove that the message exists and consumes it.
+
+In our `token-bridge` nargo project in `aztec-contracts`, under `src` there is an example `main.nr` file. Delete all the code in here and paste this to define imports and initialize the constructor:
+
+```rust
+mod util;
+mod token_interface;
+
+// Minimal implementation of the token bridge that can move funds between L1 <> L2.
+// The bridge has a corresponding Portal contract on L1 that it is attached to
+// And corresponds to a Token on L2 that uses the `AuthWit` accounts pattern.
+// Bridge has to be set as a minter on the token before it can be used
+
+contract TokenBridge {
+    use dep::aztec::{
+        context::{Context},
+        hash::{compute_secret_hash},
+        state_vars::{public_state::PublicState},
+        types::type_serialization::field_serialization::{
+            FieldSerializationMethods, FIELD_SERIALIZED_LEN,
+        },
+        types::address::{AztecAddress, EthereumAddress},
+        selector::compute_selector,
+    };
+
+    use crate::token_interface::Token;
+
+    // Storage structure, containing all storage, and specifying what slots they use.
+    struct Storage {
+        token: PublicState<Field, 1>,
+    }
+
+    impl Storage {
+        fn init(context: Context) -> pub Self {
+            Storage {
+                token: PublicState::new(
+                    context,
+                    1,
+                    FieldSerializationMethods,
+                ),
+            }
+        }
+    }
+
+    // Constructs the contract.
+    #[aztec(private)]
+    fn constructor(token: AztecAddress) {
+        let selector = compute_selector("_initialize((Field))");
+        context.call_public_function(context.this_address(), selector, [token.address]);
+    }
+```
+
+Then paste this `claim_public` function:
+#include_code claim_public /yarn-project/noir-contracts/src/contracts/token_bridge_contract/src/main.nr rust
+
+and this `mint_public_content_hash` function:
```

Comment:
uhh this code lives in `util.nr` So maybe tell people to create `util.nr` now and add this code and add1 line on why this code needs to be pasted,s something like "and lets create a helper method to compute the `contentHash` from before

Reply:
fixed this everywhere